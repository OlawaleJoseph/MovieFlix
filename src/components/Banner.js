/* eslint-disable camelcase */
import React, { useEffect, useState } from 'react';
// import PropTypes from 'prop-types';
import {
  makeStyles, Grid, Typography, Button,
} from '@material-ui/core';
import { generateRandowmNumber } from '../helpers/common';
import { urls, imageBaseUrl } from '../helpers/constants';
import fetch from '../helpers/data';

const Banner = () => {
  const [movie, setMovie] = useState(null);
  useEffect(async () => {
    const fetchMovies = async () => {
      const { data: { results } } = await fetch.get(urls.originals);
      const index = generateRandowmNumber(results.length);

      return results[index];
    };
    try {
      const data = await fetchMovies();
      setMovie(data);
    } catch (error) {
      console.log('Error occurred');
      console.log(error);
    }
  }, []);

  const bgImg = imageBaseUrl + movie?.backdrop_path;

  const styles = makeStyles(theme => ({
    bg: {
      background: `url('${bgImg}') no-repeat center`,
      backgroundSize: 'cover',
      width: '100%',
      height: '450px',
      paddingTop: theme.spacing(6),
    },
    content: {
      height: '100%',
      paddingLeft: '30px',
      color: 'white',
    },
    btnContainer: {
      marginTop: '20px',
    },
    btn: {
      marginRight: '10px',
      background: 'white',
    },
  }));

  const classes = styles();

  return (
    <div className={classes.bg}>
      <Grid container alignItems="center" className={classes.content}>
        <Grid item xs={10} md={5} lg={4}>
          <Typography variant="h4" className={classes.heading}>
            {movie?.name}
          </Typography>
          <Typography variant="body2">
            {movie?.overview}
          </Typography>
          <div className={classes.btnContainer}>
            <Button
              className={classes.btn}
              variant="outlined"
              size="large"
              color="secondary"
            >
              Play
            </Button>
            <Button
              className={classes.btn}
              variant="outlined"
              size="large"
              color="secondary"
            >
              Details
            </Button>
          </div>
        </Grid>
      </Grid>
    </div>
  );
};

// Banner.propTypes = {
//   bgImg: PropTypes.string.isRequired,
//   desc: PropTypes.string.isRequired,
//   title: PropTypes.string.isRequired,
// };

export default Banner;
