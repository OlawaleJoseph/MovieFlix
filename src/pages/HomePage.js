import React from 'react';
import Banner from '../components/Banner';
import MovieRow from '../containers/MovieRow';
import { urls } from '../helpers/constants';

const HomePage = () => (
  <div>
    <Banner />
    <MovieRow title="Popular Movies" moviesUrl={urls.moviePopular} />
    <MovieRow title="Popular Tv Shows" moviesUrl={urls.popularTv} />
    <MovieRow title="Top Rated Movies" moviesUrl={urls.topRatedMovie} />
    <MovieRow title="Top Rated Tv Shows" moviesUrl={urls.topRatedTv} />
    <MovieRow title="Upcoming" moviesUrl={urls.upcoming} />
  </div>
);

export default HomePage;
