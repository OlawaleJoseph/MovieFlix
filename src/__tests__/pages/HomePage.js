import HomePage from '../../pages/HomePage';
import Row from '../../containers/MovieRow';
import Banner from '../../components/Banner';
import shallowWrapper from '../../testSetup';

describe('Home page', () => {
  const page = shallowWrapper(HomePage);

  test('should render home page', () => {
    expect(page).toBeTruthy();
    expect(page).toMatchSnapshot();
  });

  test('should render Banner', () => {
    const movieDescription = page.find(Banner);

    expect(movieDescription).toHaveLength(1);
  });

  test('should render Moview Row', () => {
    const row = page.find(Row);

    expect(row).toHaveLength(5);
  });
});
