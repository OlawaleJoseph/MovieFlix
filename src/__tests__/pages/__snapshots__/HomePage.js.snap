// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should render Movie page 1`] = `ShallowWrapper {}`;

exports[`Home page should render home page 1`] = `ShallowWrapper {}`;

exports[`Movie Row should render Movie Row 1`] = `ShallowWrapper {}`;

exports[`Movie page should render Movie page 1`] = `ShallowWrapper {}`;
